iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift

```diff
@@ -31,134 +31,183 @@ import SalesforceSDKCore
+                            .zIndex(2.0)
+                        
+                        Button {
+                            Task {
+                                errorMessage = ""
+                                self.isAuthenticating = true
+                                
+                                // Login
+                                let result = await SalesforceManager.shared.nativeLoginManager()
+                                    .login(username: username, password: password)
+                                self.isAuthenticating = false
+                                
+                                switch result {
+                                case .invalidCredentials:
+                                    errorMessage = "Please check your username and password."
+                                    break
+                                case .invalidUsername:
+                                    errorMessage = "Username format is incorrect."
+                                    break
+                                case .invalidPassword:
+                                    errorMessage = "Invalid password."
+                                    break
+                                case .unknownError:
+                                    errorMessage = "An unknown error has occurred."
+                                    break
+                                case .success:
+                                    self.password = ""
+                                }
                             }
+                        } label: {
+                            HStack {
+                                Image(systemName: "lock")
+                                Text("Log In")
+                            }.frame(minWidth: 150)
                         }
-                    } label: {
-                        HStack {
-                            Image(systemName: "lock")
-                            Text("Log In")
-                        }.frame(minWidth: 150)
-                    }
-                    .buttonStyle(.bordered)
-                    .tint(.blue)
-                    .zIndex(2.0)
-                }.padding(.bottom, 0)
-            }
-            .frame(maxHeight: .infinity, alignment: .center)
-            .padding(.bottom, 125)
-            
-            // Fallback to webview based authentication.
-            Button("Looking for Salesforce Log In?") {
-                SalesforceManager.shared.nativeLoginManager().fallbackToWebAuthentication()
+                        .buttonStyle(.bordered)
+                        .tint(.blue)
+                        .zIndex(2.0)
+                    }.padding(.bottom, 0)
+                }
+                
+                // Other login options.
+                Button("Need to register, reset your password or login without a password?") {
+                    navigationPathObservable.navigationPath.append(.NativeRequestOtpView)
```

Comment:
NIT: I think this button either needs padding or to be broken up into multiple (icon?) buttons or something.

Reply:
Agreed.  For now, I just changed it to "Use Passcode Instead".

Reply:
[Passcode already has a specific meaning unrelated to this.](https://developer.salesforce.com/docs/atlas.en-us.mobile_sdk.meta/mobile_sdk/android_passcodes.htm)